### PR TITLE
7746 inital changes for confirmation

### LINKED ISSUE #7746: After changing the data directory for nethermind client, default jwt secret is still generated in default directory.
**Description**
In the absense of jwt-secret file and a change in default data directory, jwt-secret file is still generated in default data directory.

Note:
check for similar behaviour if data directory is specified using env or config file.

**Steps to Reproduce**
In order to replicate the behavior, please provide a detailed list of steps:
1) change the default data directory using `--dd <new_data_dir>`
2) delete previous jwt-secret
3) ensure jwt-secret is not specified in config file or as config  parameter (so it gets auto generated)

**Actual behavior**
jwt-secret file is generated in `src/Nethermind/Nethermind.Runner/keystore.`

**Expected behavior**
jwt-secret file should be generated in `<new_data_dir>/keystore`

### PR DESCRIPTION
**Pull request by ssonthal**
Fixes  #7746 

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Changes

- _List the changes_

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [ ] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [ ] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

**Comment by ssonthal** [comment]
Hey @obasekiosa , gentle reminder for your review on this one.

**Comment by LukaszRozmej** [comment]
I am not sure if we should update one path based on another or handle it deeper when resolving the path. @rubo ?

**Comment by ssonthal** [comment]
Yes, i will fix the submodule changes.

File: src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartRpc.cs [hunk 1]
```diff
@@ -30,6 +32,11 @@ public StartRpc(INethermindApi api)
         public async Task Execute(CancellationToken cancellationToken)
         {
             IJsonRpcConfig jsonRpcConfig = _api.Config<IJsonRpcConfig>();
+            IKeyStoreConfig keyStoreConfig = _api.Config<IKeyStoreConfig>();
+
+            // updating jwt-secret path based on data directory.
+            jsonRpcConfig.JwtSecretFile = String.Concat(keyStoreConfig.KeyStoreDirectory,"/jwt-secret");
```
Comment (rubo) [inline src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartRpc.cs:38]: This change actually introduces another bug disrespecting the `JsonRpc.JwtSecretFile` setting. I think we should remove the default value of the `IJsonRpcConfig.JwtSecretFile` and check for null. If null, then apply this change.

**Review by rubo — CHANGES_REQUESTED** [review]

**Comment by ssonthal** [comment]
Creating a new PR for the suggested changes as I wasn't able to undo the submodule changes - https://github.com/NethermindEth/nethermind/pull/7765

**Pull request closed without merging**